OpenSim/Common/Component.h

```diff
@@ -1554,6 +1554,7 @@ template <class T> friend class ComponentMeasure;
             _connectorsTable[connector.getName()] = ix;
         }
     }
+protected:
```

Comment:
Thanks @aseth1 and @chrisdembia I moved the function within the header and beefed up the documentation to indicate when it should be called. I'd prefer to leave the renaming for another issue as we don't use (sub) anywhere else, and also so it doesn't get entangled with other changes.

@chrisdembia The method is now called from ModelComponent::connectToModel method rather than a Component method. The testComponentInterface test case uses Compoenents.rather than ModelComponents, I understand they maybe merged in the future. The test case will need to be redone once we have a complete design for the Environment functionality.